include/seqan3/range/view/single_pass_input.hpp

```diff
@@ -0,0 +1,402 @@
+// Iterator for single pass input view.
+//-----------------------------------------------------------------------------
+
+namespace seqan3::detail
+{
+/*!\brief An input_iterator over the associated range.
+ * \implements input_iterator_concept
+ * \ingroup view
+ * \tparam view_type The type of the associated type.
+ *
+ * This iterator reduces every iterator type of the associated view to an single pass input iterator.
+ */
+template <typename view_type>
+class single_pass_input_iterator<single_pass_input_view<view_type>>
+{
+    //!\brief The pointer to the associated view.
+    using base_iterator_t = typename single_pass_input_view<view_type>::urng_iterator_type;
+    //!\brief The pointer to the associated view.
+    single_pass_input_view<view_type> * view_ptr{};
+
+    //!\ Friend declaration to give seqan3::detail::single_pass_input_sentinel access to members of this class.
+    template <typename sentinel_t>
+    friend class single_pass_input_sentinel;
+
+public:
+
+    /*!\name Member types
+     * \{
+     */
+    //!\brief Difference type.
+    using difference_type   = difference_type_t<base_iterator_t>;
+    //!\brief Value type.
+    using value_type        = value_type_t<base_iterator_t>;
+    //!\brief Pointer type.
+    using pointer           = std::add_pointer_t<value_type>;
+    //!\brief Reference type.
+    using reference         = reference_t<base_iterator_t>;
+    //!\brief Iterator category.
+    using iterator_category = std::input_iterator_tag;
+    //!\}
+
+    /*!\name Construction, destruction and assignment.
+     * \{
+     */
+    //!\brief Default construction.
+    single_pass_input_iterator() = default;
+    //!\brief Copy construction.
+    constexpr single_pass_input_iterator(single_pass_input_iterator const & rhs) = default;
+    //!\brief Move construction.
+    constexpr single_pass_input_iterator(single_pass_input_iterator && rhs) = default;
+    //!\brief Copy assignment.
+    constexpr single_pass_input_iterator & operator=(single_pass_input_iterator const & rhs) = default;
+    //!\brief Move assignment.
+    constexpr single_pass_input_iterator & operator=(single_pass_input_iterator && rhs) = default;
+    //!\brief Destruction.
+    ~single_pass_input_iterator() = default;
+
+    //!\brief Constructing from the underlying seqan3::single_pass_input_view.
+    single_pass_input_iterator(single_pass_input_view<view_type> & view) noexcept : view_ptr{&view}
+    {}
+    //!\}
+
+    /*!\name Access operations.
+     * \{
+     */
+    //!\brief Dereferences the cached iterator.
+    reference operator*() const noexcept
+    {
+        return *cached();
+    }
+    //!\}
+
+    /*!\name Operations
+     * \{
+     */
+    //!\brief Pre-increment.
+    single_pass_input_iterator & operator++() noexcept
+    {
+        ++cached();
+        return *this;
+    }
+
+    //!\brief Post-increment.
+    void operator++(int) noexcept
+    {
+        ++(*this);
+    }
+    //!\}
```

Comment:
For the other iteerators these are together in a group just called "Iterator operations"